agent/engine/task_manager.go

```diff
@@ -148,6 +149,18 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO: Add new task resources provisioned state ?
+			err := mtask.SetupPlatformResources()
+			if err != nil {
+				seelog.Errorf("Unable to set up resources for task %s: %v", mtask.Task.Arn, err)
+				// If task cgroup creation fails, set task status to stopped
+				// with valid reason
+				mtask.SetKnownStatus(api.TaskStopped)
+				mtask.SetDesiredStatus(api.TaskStopped)
+				mtask.engine.emitTaskEvent(mtask.Task, taskUnableToCreateCgroup)
+				return
```

Comment:
We shouldn't return here, I think this will cause the task status not reported to backend? You just need to set the desired status to be stopped.